omod/src/main/java/org/openmrs/module/owa/impl/DefaultAppManager.java

```diff
@@ -92,13 +93,21 @@ public void installApp(File file, String fileName, String rootPath) throws IOExc
 
 				// ---------------------------------------------------------------------
 				// Delete if app is already installed
+				// If app specified 'deployed.owa.name', use it instead of default name based on package name
 				// ---------------------------------------------------------------------
-				String dest = getAppFolderPath() + File.separator + fileName.substring(0, fileName.lastIndexOf('.'));
 
-				if (getApps().contains(app)) {
+				String deployedName = fileName.substring(0, fileName.lastIndexOf('.'));
+
+				if(StringUtils.isNotBlank(app.getDeployedName())){
+					deployedName = app.getDeployedName();
+					//delete app deployed in the same directory
+					deleteApp(app.getDeployedName());
```

Comment:
You would just use deployedName instead of calling app.getDeployedName() again. :)